hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngine.java

```diff
@@ -99,4 +100,12 @@ QueryResult queryLocalPartitions(String name, Predicate predicate, IterationType
      * @return {@link SortedQueryResultSet}
      */
     Set queryAllPartitionsWithPagingPredicate(String mapName, PagingPredicate pagingPredicate, IterationType iterationType);
+
+
+    /**
+     * Returns {@link QueryOptimizer} for this engine.
+     *
+     * @return {@link QueryOptimizer} for this engine.
+     */
+    QueryOptimizer getQueryOptimizer();
```

Comment:
I'm not super excited about this. optimization should be an internal concern. is there any other option?